File: src/cursor.c
#include <stdlib.h>
#include "cursor.h"

Cursor *init_cursor(void) {
	Cursor *cursor = malloc(sizeof(Cursor));
	cursor->canClick = 1;
	cursor->icon = NULL;
	cursor->xPos = 0;				cursor->yPos = 0;
	cursor->xOld = cursor->xPos;	cursor->yOld = cursor->yPos;
	return cursor;
}

File: src/cursor.lua
local cursor = {
	canclick = true,
	icon = "  ",
	yPos = 0, xPos = 0,
	yOld = 0, xOld = 0,
}

function cursor:new(screen)
	local obj = {screen = screen}
	return setmetatable(obj, {__index = self})
end
